backend/services/soft_deletion.py

```
@@ -48,27 +48,24 @@ def soft_delete_scenario(
-            # Archive related user progress
-            self._archive_user_progress_for_scenario(scenario_id, reason)
+            # Archive related user progress (soft delete only)
+            self._soft_delete_user_progress_for_scenario(scenario_id, reason)
             
-            # Immediately clean up the archived data (permanent deletion)
-            self._immediate_cleanup_scenario_archives(scenario_id)
-            
-            # Hard delete the scenario and all related records
-            self._hard_delete_scenario(scenario_id)
+            # Soft delete the scenario
+            scenario.deleted_at = datetime.utcnow()
```

Ruff (0.14.1)

`datetime.datetime.utcnow()` used

(DTZ003)